### PR TITLE
Retry S3 InternalError and RequestTimeout errors

### PR DESCRIPTION
**Pull request by edolstra**
This fixes "Unable to parse ExceptionName: InternalError Message: We encountered an internal error. Please try again." and "Unable to parse ExceptionName: RequestTimeout Message: Your socket connection to the server was not read from or written to within the timeout period. Idle connections will be closed." exceptions when doing an S3 request. Both errors now cause the request to be retried.

### CODE REVIEW

**Comment by edolstra** [comment]
Updated to also handle the `SlowDown` ("please reduce your request rate") exception.

File: aws-cpp-sdk-s3/include/aws/s3/S3Errors.h [hunk 1]
```diff
@@ -54,12 +54,13 @@ enum class AWS_S3_API S3Errors
   NO_SUCH_KEY,
   NO_SUCH_UPLOAD,
   OBJECT_ALREADY_IN_ACTIVE_TIER,
-  OBJECT_NOT_IN_ACTIVE_TIER
+  OBJECT_NOT_IN_ACTIVE_TIER,
+  SLOWDOWN
```
Comment (JonathanHenson) [inline aws-cpp-sdk-s3/include/aws/s3/S3Errors.h]: This value is autogenerated from the s3 model, can you post the exact string you recieved from S3 so I can properly update their model. Then this change will go out automatically.

Comment (edolstra) [inline aws-cpp-sdk-s3/include/aws/s3/S3Errors.h]: The AWSError exception thrown by aws-sdk-cpp has the following message:
```
Unable to parse ExceptionName: SlowDown Message: Please reduce your request rate.
```

Comment (JonathanHenson) [inline aws-cpp-sdk-s3/include/aws/s3/S3Errors.h]: Can you upload the log that shows this request and response for me?

Comment (edolstra) [inline aws-cpp-sdk-s3/include/aws/s3/S3Errors.h]: The request:
```
PUT /lb-s3-speed-test/pages/39257486e3e1d089
authorization: AWS4-HMAC-SHA256 Credential=.../eu-west-1/s3/aws4_request, SignedHeaders=content-length;content-type;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=5f4d41f2184e255fb246c46ae27babd733876b640e4d73f9e4dd611bc28dcb50
content-length: 32768
content-type: binary/octet-stream
host: s3-eu-west-1.amazonaws.com
user-agent: aws-sdk-cpp/0.10.7 Linux/4.4.14 x86_64
x-amz-content-sha256: 1a6a1d5206e70c3952eabd63fc3f375237b486dde426d4d110089413359b5847
x-amz-date: 20160715T130851Z
```

The response:
```
HTTP/1.1 503 Slow Down
x-amz-request-id: 010021009046B59A
x-amz-id-2: kOGUvPllYfS9Efd6x7C7ZEXBQZYnymj7UBDatZ/+/2tNB2fTl2H2tRY8YIcsmyQEJ7NCGSalXm8=
Content-Type: application/xml
Transfer-Encoding: chunked
Date: Fri, 15 Jul 2016 13:08:51 GMT
Server: AmazonS3

<?xml version="1.0" encoding="UTF-8"?>
<Error>
    <Code>SlowDown</Code>
    <Message>Please reduce your request rate.</Message>
    <RequestId>010021009046B59A</RequestId>
    <HostId>kOGUvPllYfS9Efd6x7C7ZEXBQZYnymj7UBDatZ/+/2tNB2fTl2H2tRY8YIcsmyQEJ7NCGSalXm8=</HostId>
</Error>
```

Comment (JonathanHenson) [inline aws-cpp-sdk-s3/include/aws/s3/S3Errors.h]: SlowDown has been added as a core error that will be retried.

**Comment by JonathanHenson** [comment]
This has been updated and deployed (months ago) sorry for the delay in updating you.

**Comment by prgtrdr** [comment]
I just received this error ("Unable to parse ExceptionName: InternalError Message: We encountered an internal error. Please try again.") using v1.0.133 of AWSSDKCPP with an Ubuntu EC2 image.  Is there something I should do in my code to retry?  Is there a way to gracefully recover?
image

**Pull request closed without merging**